hits_cpp/hits.cpp

```diff
@@ -165,14 +167,62 @@ MemeRatingMap getRatings(Memes& memes, MemeShufflesMap& shuffles, bool verbose)
             cout << "Motif " << memeIter->first << ", Total shuffles: " << memeIter->second.size() <<
                 ", Rank: " << rank << ", Score: " << score << endl;
         }
+        */
+        int add=memeIter->second.size();
+        if (percent!=0.0){
+            add++;
+        }
+        int shuffleArray[add];
+        int count=0;
+        int max=0;
+        //create list of the suffle hits
+        while (shuffleIter != shuffleEnd) {
+                shuffleArray[count]=shuffleIter->getHitCount();
+                if (shuffleIter->getHitCount()>max){
+                    max=shuffleIter->getHitCount();
+                }
+                shuffleIter++;
+                count++;
+        }
+        if (add!=memeIter->second.size()){
+            shuffleArray[memeIter->second.size()]=(int)((float)max*percent)+max;
+        }
+        //sort the list 
+        int n = sizeof(shuffleArray) / sizeof(shuffleArray[0]);
+        sort(shuffleArray, shuffleArray + n);
+        
+        int shuffleHitStart=shuffleArray[0];
+        int shuffleHitEnd=shuffleArray[n-1];
+        float score;
+        if (hits==0){
+            score=0.0;
+        }
+        else if (hits>shuffleHitEnd){
+            score=1.0;
+        }
+        else if (hits<shuffleHitStart){
+            score=0.0;
+        }else{
+            for (int place=0; place<n;place++){
+                if (hits <=shuffleArray[place]){
+                    float min_1=(float)shuffleArray[place-1];
+                    float max_1=(float)shuffleArray[place];
+                    float z_maen=((float)hits-min_1)/(max_1-min_1);
+                    float p=(float)place/(float)n;
+                    score=(z_maen/(float)n)+p;
+                    break;
+                }
+			} 
+        }
+        cout<<"score"<<score<<endl;
```

Comment:
Do we always want to print this or only if `verbose` is true?
See previous implementation, printing just the score is not clear enough